api/src/main/scala/dcos/metronome/api/v1/controllers/JobSpecController.scala

```diff
@@ -69,23 +69,22 @@ class JobSpecController(
 
   def deleteJob(id: JobId, stopCurrentJobRuns: Boolean) = measured {
     AuthorizedAction.async { implicit request =>
-      def deleteJobSpec(jobSpec: JobSpec): Future[Result] = async {
-        val runs = await(jobRunService.activeRuns(id))
-        if (runs.nonEmpty && !stopCurrentJobRuns) {
-          Conflict(ErrorDetail("There are active job runs. Override with stopCurrentJobRuns=true"))
-        } else {
-          await {
-            Future.sequence(runs.map(run => jobRunService.killJobRun(run.jobRun.id))).recover {
-              case _: JobSpecDoesNotExist => NotFound(UnknownJob(id))
-            }
-          }
-          await {
-            jobSpecService.deleteJobSpec(id).map(Ok(_)).recover {
-              case _: JobSpecDoesNotExist => NotFound(UnknownJob(id))
+      def deleteJobSpec(jobSpec: JobSpec): Future[Result] = {
+        jobRunService.activeRuns(id).flatMap {
+          case _ :: _ if !stopCurrentJobRuns =>
+            Future.successful(Conflict(ErrorDetail("There are active job runs. Override with stopCurrentJobRuns=true")))
+          case runs =>
+            val f = for {
+              _ <- Future.sequence(runs.map(run => jobRunService.killJobRun(run.jobRun.id)))
+              deleteRes <- jobSpecService.deleteJobSpec(id)
+            } yield Ok(deleteRes)
```

Comment:
impressive... this terse and we need to make it as readable as possible.... can we rename or fully name `deleteRes`

Reply:
Yes. Just to give context, I was removing a warn because of the two awaits that were not going anywhere (just generated side effect or an exception). Tried to refactor in a way to make the later step be a for comprehension noting that they had to run in order and either could fail. 

But yes will rename it

Reply:
renamed to `deletedJobSpec`